querydsl-jpa/src/main/java/com/querydsl/jpa/NativeSQLSerializer.java

```diff
@@ -194,12 +194,12 @@ public void visitConstant(Object constant) {
                 first = false;
             }
             append(")");
-        } else if (!getConstantToNamedLabel().containsKey(constant)) {
-            String constLabel = String.valueOf(getConstantToNamedLabel().size() + 1);
-            getConstantToNamedLabel().put(constant, constLabel);
+        } else if (!getConstantToAllLabels().containsKey(constant)) {
+            final Integer constLabel = getConstantToNumberedLabel().size() + 1;
```

Comment:
final is not "worth the syntactic overhead", had this one out with Brian Goetz the main man here https://stackoverflow.com/questions/34277511/why-are-java-method-constructor-parameters-not-final-by-default